### PR TITLE
fix(build): remove ungated  >= Swift 5.4 features to allow building on Xcode 12

### PR DESCRIPTION
**Pull request by atierian**
*Issue #, if available:*
https://github.com/aws-amplify/amplify-flutter/issues/1512

*Description of changes:*
Fixes build error on Xcode 12.4

*Check points: (check or cross out if not relevant)*

- [ ] ~Added new tests to cover change, if needed~
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift [hunk 1]
```diff
@@ -119,7 +119,7 @@ extension ModelSchema {
     func createIndexStatements() -> String {
         var statement = ""
         for index in indexes {
-            if case let .index(fields, name) = index, let name = name {
+            if case let .index(fields, name?) = index {
```
Comment (royjit) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift:122]: Does this make `name` an optional whereas earlier it was required to have a value?

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift [hunk 2]
```diff
@@ -118,10 +118,13 @@ extension ModelSchema {
     /// Create SQLite indexes corresponding to secondary indexes in the model schema
     func createIndexStatements() -> String {
         var statement = ""
-        for index in indexes {
-            if case let .index(fields, name) = index, let name = name {
-                statement += CreateIndexStatement(modelSchema: self, fields: fields, indexName: name).stringValue
-            }
+        for case let .index(fields, name?) in indexes {
```
Comment (diegocstn) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift:121]: can we add a `where` clause and filter out indexes with a `nil` name?

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift (hunk 1 above)
Comment (atierian) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift]: Nope, this only executes if `name != nil`.

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift (hunk 2 above)
Comment (atierian) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift:121]: That's what the trailing `?` does.

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift (hunk 1 above)
Comment (lawmicha) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift]: @thisisabhash may have some context here

Comment (atierian) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift]: Nope, `CreateIndexStatement(modelSchema:fields:indexName:)` takes a non-optional `String`. The trailing `?` makes use of the [Optional Pattern Matching Operator](https://docs.swift.org/swift-book/ReferenceManual/Patterns.html#ID520) to safely unwrap and only execute if `name != nil`

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift (hunk 2 above)
Comment (diegocstn) [reply AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/ModelSchema+SQLite.swift:121]: Ah, nvm I missed that.

**Review by diegocstn — APPROVED** [review]

**Comment by atierian** [comment]
@thisisabhash I'd love your review on this since you have the most context around this code.

**Pull request merged**